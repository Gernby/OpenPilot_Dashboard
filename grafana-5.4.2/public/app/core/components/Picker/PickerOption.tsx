import React from 'react';
import { components } from 'react-select';
import { OptionProps } from 'react-select/lib/components/Option';

// https://github.com/JedWatson/react-select/issues/3038
interface ExtendedOptionProps extends OptionProps<any> {
  data: any;
}

export const PickerOption = (props: ExtendedOptionProps) => {
  const { children, data, className } = props;
  return (
    <components.Option {...props}>
      <div className={`description-picker-option__button btn btn-link ${className}`}>
        {data.avatarUrl && <img src={data.avatarUrl} alt={data.label} className="user-picker-option__avatar" />}
        {children}
      </div>
    </components.Option>
  );
};

export default PickerOption;
