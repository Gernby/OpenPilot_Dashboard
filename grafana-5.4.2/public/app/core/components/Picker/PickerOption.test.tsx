import React from 'react';
import renderer from 'react-test-renderer';
import PickerOption from './PickerOption';

const model = {
  cx: jest.fn(),
  clearValue: jest.fn(),
  onSelect: jest.fn(),
  getStyles: jest.fn(),
  getValue: jest.fn(),
  hasValue: true,
  isMulti: false,
  options: [],
  selectOption: jest.fn(),
  selectProps: {},
  setValue: jest.fn(),
  isDisabled: false,
  isFocused: false,
  isSelected: false,
  innerRef: null,
  innerProps: null,
  label: 'Option label',
  type: null,
  children: 'Model title',
  data: {
    title: 'Model title',
    avatarUrl: 'url/to/avatar',
    label: 'User picker label',
  },
  className: 'class-for-user-picker',
};

describe('PickerOption', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<PickerOption {...model} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
