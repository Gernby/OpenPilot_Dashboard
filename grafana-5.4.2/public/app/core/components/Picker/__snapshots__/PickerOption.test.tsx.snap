// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PickerOption renders correctly 1`] = `
<div>
  <div
    className="description-picker-option__button btn btn-link class-for-user-picker"
  >
    <img
      alt="User picker label"
      className="user-picker-option__avatar"
      src="url/to/avatar"
    />
    Model title
  </div>
</div>
`;
  